samtranslator/model/intrinsics.py

```diff
@@ -146,6 +146,23 @@ def is_instrinsic(input):
     return False
 
 
+def is_instrinsic_fn(input):
+    """
+    Checks if the given input is an intrinsic function dictionary. Intrinsic function is a dictionary with single
+    key that is the name of the intrinsics.
+
+    :param input: Input value to check if it is an intrinsic (all except Ref)
+    :return: True, if yes
+    """
+
+    if input is not None and isinstance(input, dict) and len(input) == 1:
+
+        key = list(input.keys())[0]
+        return key == "Condition" or key.startswith("Fn::")
```

Comment:
If `!Sub` is given in the template does that get resolved to `Fn::Sub` at this point?

Reply:
sorry this change snuck in. Not using this new fn anywhere so will be getting rid of it